pagefaultgames/pokerogue
# [Bug] fix #5034 removed unnecessary caught data block for certain forms

## What are the changes the user will see?

Caught data will now be tracked correctly. When a trainer uses a Mega form you have unlocked already, it won't show as a grey ball icon anymore. Currently a user can only find these forms when they use RogueDex google sheet. In the future this would show up in the in-game pokedex too.

## Why am I making these changes?


I am making these changes so the collectors can actually collect all the remaining forms and so users finally have a way to get rid of those grey ball icons on some mega form trainer pokemon. Issue included in title.
fixes #5034 

## What are the changes from a developer perspective?

Removed a list of form names from pokemon-species.ts. Removed an if-statement from game-data.ts. The no starter form checks are done by configuring isStarterSelectable in PokemonForm, which is already done for every pokemon in the game. The list that I removed is just an old bit of code that was never removed.

## Screenshots/Videos

Caught attribute of Palkia using the data and session of the referenced issue is now 405n, that is 110010101 in binary, where the first two 1s are the caught forms. Before the change this was only the base form, now the second Origin form is also stored in caught data. Any trainer that would use Palkia Origin would no longer have a grey ball icon when I face them in the future. (the form is still not available in starter select, see section above this)
![image](https://github.com/user-attachments/assets/78541d6a-4b0f-431a-8e6a-8e8bb3b16a6d)
And proof that the form is still not selectable after removing the obsolete code: ![image](https://github.com/user-attachments/assets/6df2bd58-6441-4d00-a637-0020ce5d07a0)


## How to test the changes?

See the issue number. Import data, import session. Continue, beat the wave. Go to check team. Disable and Activate Lustrous Globe.
game-data.ts - setPokemonSpeciesCaught should now set formIndex 1 in caught data, resulting in a caught data value of 405n.

## Checklist

Are there any localization additions or changes? If so:
